m4/ax_c_mallinfo.m4

```diff
@@ -0,0 +1,21 @@
+# AC_C_MALLINFO
+# -------------
+# Define HAVE_C_MALLINFO if mallinfo() works.
+AN_IDENTIFIER([mallinfo], [AC_C_MALLINFO])
+AC_DEFUN([AC_C_MALLINFO],
+[AC_CACHE_CHECK([for mallinfo], ac_cv_c_mallinfo,
+[AC_LINK_IFELSE(
+   [AC_LANG_SOURCE(
+      [[#include <malloc.h>
+        int main(int argc, char **argv) {
+          struct mallinfo mi = mallinfo();
+          printf("%d%d%d\n", mi.uordblks, mi.hblkhd, mi.arena);
```

Comment:
Should be something like:

instead of the AC_LANG_SOURCE use the following:

AC_LANG_PROGRAM(
    [[#include <malloc.h>]],
    [[
        struct mallinfo mi = mallinfo();
        /* make sure that fields exists */
        mi.uordblks = 0;
        mi.hblkhd = 0;
        mi.arena = 0;
    ]]
)

AC_LANG_PROGRAM should be used in almost all cases, AC_LANG_SOURCE only in rare cases.